### PR TITLE
Fixing the sentimental analysis part

### PR DESCRIPTION
**Pull request by segfal**
A similar PR may already be submitted!
Please search among the [Pull request](../) before creating one.

Thanks for submitting a pull request! Please provide enough information so that others can review your pull request:

For more information, see the `CONTRIBUTING` guide.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

** The Sentimental Analysis part has been simplified. It would fix the few errors that were popping up.**

- * First error was that the sentimental variable was not taking any of the arrays cause of a bad index. I fixed it by changing the index to the "created_by" timestamp so it can not give that error anymore and it can take the timestamps to graph the data.
- Second is the graph showing nothing, i basically turned the interval down to 10 seconds instead of 5T which i think is 5 minutes, and the graphs started to pop up.
- Third, I turned the sentimental analysis part into a class so we don't have to do alot of repetition, everything would be more seamless and the code would look cleaner
- Finally, I added argv into the graph method, so you dont have to just input one hashtag, you can input multiple hashtags and get data from  multiple hashtags.

* **What is the current behavior?** (You can also link to an open issue here)
* It spits out a graph on main.py and provides the scores as well.

* **What is the new behavior (if this is a feature change)?**
* Theres a function that takes multiple hashtags instead of just one and in the graph function , I reccomend for the interval you put it as graph(interval=time,hashtags) so the function will know that you added the interval

Graphs

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Basically I just deleted code that was redundant , there doesn't need to be any changes. no extra libraries or stuff like that, but if you want some nice graphs i recommend the plotly library as their graphs are more interactive.

* **Other information**:

https://i.imgur.com/XQavA8e.png

Explain the **motivation** for making this change. What existing problem does the pull request solve?

**Test plan (required)**

Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

**Code formatting**

**Closing issues**

Fixes issue #15

### CODE REVIEW

**Review by kaustav202 — CHANGES_REQUESTED** [review]
It's good overall.. the sentiment part is done, but need to restore the functionalities of the main file that you removed ( total mentions plot, average plot and proportion of total tweets )

**Comment by kaustav202** [comment]
Also clean up the code.. redundant newlines, comments etc

**Comment by segfal** [comment]
@kaustav202  Just did the change, I restored the code from main.py and cleaned up sentimental_analysis.py

File: app/main.py [hunk 1]
```diff
@@ -11,6 +12,82 @@
 ds_tweets = get_tweets()
 
 
+
+
 sentiments  = sentiment_analysis.SentimentalAnalysis(ds_tweets=ds_tweets)
 sentiments.set_hashtag_data('5T','#apple','#google')
 sentiments.graph()
```
Comment (kaustav202) [inline app/main.py:19]: The sentiment plots should be after the other plots ( at the end )

File: app/main.py [hunk 2]
```diff
@@ -11,6 +12,82 @@
...
+
+
+
+
+
+def check_word_in_tweet(word, data):
```
Comment (kaustav202) [inline app/main.py:25]: This function can simply be imported from other file, no need to define again

**Review by kaustav202 — CHANGES_REQUESTED** [review]
Just finish the minor changes and we are good to go

**Comment by segfal** [comment]
I just did a few fixes, i called sentiment from sentiment_analysis.py and added sentiment.graph() to the bottom

File: app/main.py [hunk 3]
```diff
@@ -3,34 +3,32 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (kaustav202) [inline app/main.py:20]: redundant space

File: app/main.py [hunk 4]
```diff
@@ -3,34 +3,32 @@
...
 
 ds_tweets = get_tweets()
 
-python = check_word_in_tweet('#google', ds_tweets)
+python = sentiment.get_hashtag_data()['#google']
 
 # Find mentions of #rstats in all text fields
-rstats = check_word_in_tweet('#apple', ds_tweets)
-
+rstats = sentiment.get_hashtag_data()['#apple']
```
Comment (kaustav202) [inline app/main.py:31]: Same here

File: app/main.py [hunk 5]
```diff
@@ -46,9 +44,9 @@ def check_word_in_tweet(word, data):
 
 
 
-ds_tweets['google'] = check_word_in_tweet('#google', ds_tweets)
+ds_tweets['google'] = sentiment.get_hashtag_data()['#google']
```
Comment (kaustav202) [inline app/main.py:47]: Here too

File: app/main.py [hunk 6]
```diff
@@ -46,9 +44,9 @@ def check_word_in_tweet(word, data):
...
 
-ds_tweets['apple'] = check_word_in_tweet('#apple', ds_tweets)
+ds_tweets['apple'] = sentiment.get_hashtag_data()['#apple']
```
Comment (kaustav202) [inline app/main.py:49]: As well

**Review by kaustav202 — CHANGES_REQUESTED** [review]
No need to make any changes to sentiment_analysis.py. just correct the mistakes the main.py

> Import the check_word_in_tweet() function in main and replace where I've pointed out

- [ ] Replace the incorrect get_hashtag_data() with check_word_in_tweet()
- [ ] Remove redundant space

**Comment by kaustav202** [comment]
The main is computing meta metrics on the bulk so we are interested in only count/volume here not sentiments. The two plots are computing 1. total counts of keywords over an interval and 2. average counts of keyword in a particular interval , additionally we are printing the proportion of tweets containing each keyword in lines 33 and 36.

That was about the main files' working. I hope it is clear to you

**Comment by kaustav202** [comment]
@segfal  are you working on it or shall I add a commit myself?

File: app/main.py (hunk 6 above)
Comment (segfal) [reply app/main.py:49]: For check word in tweets should i call the function from sentimental analysis? or have it on main? my plan was to put it inside sentimental analysis.py and call it from there

**Comment by kaustav202** [comment]
> 



> @kaustav202 Ill work on it right now, just to confirm. I will remove the changes from main.py and leave it back to what it originally was?

Yes that's right..

File: app/main.py (hunk 6 above)
Comment (kaustav202) [reply app/main.py:49]: I guess import the whole class and instantiate it in main. then call check_word_ function

You could do it like this

```
from sentiment_analysis import SentimentAnalysis

sentiment = SentimentAnalysis(ds_tweets=get_tweets())

sentiment.set_hashtag_data('5T','#apple','#google') 

check_fun = sentiment.check_word_in_tweet 

// then just replace get_hashtag with function call of check_fun() at all places
```

Then fianlly remove the instantiation in sentiment_analysis.py

Comment (kaustav202) [reply app/main.py:49]: Also, it would be nice to rename the class to SentimentAnalysis instead of SentimentalAnalysis

**Comment by segfal** [comment]
> > 
> 
> > @kaustav202 Ill work on it right now, just to confirm. I will remove the changes from main.py and leave it back to what it originally was?
> 
> Yes that's right..

So i just put main.py back to its original form, the only thing i added was instansiate the sentiment inside main, and for the check word function, it will be called from the class instead of creating a whole new function.

File: app/main.py (hunk 6 above)
Comment (segfal) [reply app/main.py]: Named it to SentimentAnalysis and instantiated it inside main

**Review by kaustav202 — APPROVED** [review]
Spot on this time

**Comment by kaustav202** [comment]
great work and congratulations @segfal
